galasa-managers-parent/galasa-managers-zos-parent/dev.galasa.zos.manager/src/main/java/dev/galasa/zos/internal/ZosPortResourceMonitor.java

```diff
@@ -0,0 +1,93 @@
+/*
+* Copyright contributors to the Galasa project 
+*/
+package dev.galasa.zos.internal;
+
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import dev.galasa.framework.spi.IConfigurationPropertyStoreService;
+import dev.galasa.framework.spi.IDynamicStatusStoreService;
+import dev.galasa.framework.spi.IFramework;
+import dev.galasa.framework.spi.IResourceManagement;
+
+public class ZosPortResourceMonitor implements Runnable {
+	
+	private final Log logger = LogFactory.getLog(this.getClass());
+
+	private final IFramework 				 framework;
+	private final IResourceManagement 		 resourceManagement;
+	private final IDynamicStatusStoreService dss;
+	
+	public ZosPortResourceMonitor(IFramework framework, IResourceManagement resourceManagement, IDynamicStatusStoreService dss, ZosResourceManagement zosResourceManagement, 
+										IConfigurationPropertyStoreService cps) {
+		this.framework = framework;
+		this.resourceManagement = resourceManagement;
+		this.dss = dss;
+        this.logger.info("zOS port provisioning resource monitor initialised");
+    }
+	
+	@Override
+	public void run() {
+		logger.info("Monitoring for z/OS ports held up in allocated status");
+		
+		try {
+			
+			// Get the list of ports allocated in the DSS
+			Map<String, String> zosPortsInDss = dss.getPrefix("zosport");
+			
+			Set<String> allActiveRuns = this.framework.getFrameworkRuns().getActiveRunNames();
+			
+			// Iterate through all z/OS ports stored in the DSS
+			for (String prefix : zosPortsInDss.keySet()) {
+	
+				// Delete DSS port allocation to run not active
+				if (!allActiveRuns.contains(zosPortsInDss.get(prefix))) {
+					String removeRun = zosPortsInDss.get(prefix);
+					String removePort = prefix.split("\\.")[1];
+					
+					logger.info("Found inactive run " + removeRun + ". Freeing port " + removePort);
```

Comment:
Dont message about the run, not your job, message youve found a dirty port on image X